central/debug/service/service.go

```diff
@@ -884,6 +910,41 @@ func (s *serviceImpl) getDiagnosticDumpWithCentral(w http.ResponseWriter, r *htt
 	s.writeZippedDebugDump(r.Context(), w, filename, opts)
 }
 
+func (s *serviceImpl) getDeleRegConfigs(ctx context.Context) (interface{}, error) {
+	adminCtx := sac.WithGlobalAccessScopeChecker(
+		ctx,
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS),
+			sac.ResourceScopeKeys(resources.Administration),
+		),
+	)
+
+	config, exists, err := s.deleRegConfigDS.GetConfig(adminCtx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get delegated registry config: %w", err)
+	}
+
+	if !exists {
+		return map[string]string{
+			"message": "delegated scanning configs unavailable",
+		}, nil
+	}
+
+	r := delegatedRegistryConfig{
+		EnabledFor:       config.EnabledFor.String(),
+		DefaultClusterID: config.DefaultClusterId,
+	}
+
+	for _, rg := range config.Registries {
+		r.Registries = append(r.Registries, delegatedRegistry{
+			Path:      rg.Path,
+			ClusterID: rg.ClusterId,
+		})
+	}
+
+	return r, nil
```

Comment:
Is creating the new type and doing this conversion necessary? Any reason why not returning the object directly?

```suggestion
	return config, nil
```